Move Dimension Filters in Report/Filters Folder.

ankitrox opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #7634 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

hussain-t commented on includes/Modules/Analytics_4/Report/Filters/String_Filter.php:
```diff
@@ -1,22 +1,22 @@
+ * Class Google\Site_Kit\Modules\Analytics_4\Report\String_Filter
  *
- * @package   Google\Site_Kit\Modules\Analytics_4\Report\Dimension_Filter
+ * @package   Google\Site_Kit\Modules\Analytics_4\Report
```
Same here?

hussain-t commented on tests/phpunit/integration/Modules/Analytics_4/Report/Dimension_Filter/In_List_FilterTest.php:
```diff
@@ -10,7 +10,7 @@
 namespace Google\Site_Kit\Tests\Modules\Analytics_4\Report\Dimension_Filter;
 
-use Google\Site_Kit\Modules\Analytics_4\Report\Dimension_Filter\In_List_Filter;
+use Google\Site_Kit\Modules\Analytics_4\Report\Filters\In_List_Filter;
```
This test should be moved to the `Filters` directory.

hussain-t commented on tests/phpunit/integration/Modules/Analytics_4/Report/Dimension_Filter/String_FilterTest.php:
```diff
@@ -10,7 +10,7 @@
 namespace Google\Site_Kit\Tests\Modules\Analytics_4\Report\Dimension_Filter;
 
-use Google\Site_Kit\Modules\Analytics_4\Report\Dimension_Filter\String_Filter;
+use Google\Site_Kit\Modules\Analytics_4\Report\Filters\String_Filter;
```
This test should be moved to the `Filters` directory. And this folder should be removed.

hussain-t reviewed (changes requested):
Thanks, @ankitrox. Great work so far. However, the changes are incomplete and not yet ready to proceed. The `tests/phpunit/integration/Modules/Analytics_4/Report/Dimension_Filter` tests haven’t been moved to the  `tests/phpunit/integration/Modules/Analytics_4/Report/Filters` folder, and the `tests/phpunit/integration/Modules/Analytics_4/Report/Dimension_Filter` folder should be removed. Please take a look at my comments.

ankitrox commented:
@hussain-t Thanks for the review and sorry that I missed to move tests in the `Filters` directory.

I've now updated the package names and moved files as suggested.

Over to your for reviewing the changes.

Thanks

techanvil commented on includes/Modules/Analytics_4/Report/Filters/Filter.php:
```diff
@@ -0,0 +1,32 @@
+/**
+ * Interface for a filter class.
+ *
+ * @since 1.106.0
```
Let's add another `@since` to help track the file move. I'll apply this myself to save the round

```suggestion
 * @since 1.106.0
 * @since n.e.x.t Moved from `Analytics_4/Report/Dimension_Filters` to `Analytics_4/Report/Filters`  for use with both dimensions and metrics.
```

techanvil commented on includes/Modules/Analytics_4/Report/Filters/String_Filter.php:
```diff
@@ -1,22 +1,22 @@
- * Class for parsing the dimension string filter.
+ * Class for parsing the string filter.
  *
  * @since 1.106.0
```
As above:

```suggestion
 * @since 1.106.0
 * @since n.e.x.t Moved from `Analytics_4\Report\Dimension_Filters` to `Analytics_4\Report\Filters` for use with both dimensions and metrics.
```

techanvil reviewed (approved):
LGTM, thanks @ankitrox!

Please note I made a couple of minor tweaks myself so as to save the round trip.

I also confirm that the failing E2E tests are not related to the changes in this PR.
